pkg/controllers/deprovisioning/multinodeconsolidation.go

```diff
@@ -140,23 +140,23 @@ func (m *MultiMachineConsolidation) firstNMachineConsolidationOption(ctx context
 // consolidated if the list of replacement instance types include one of the instance types that is being removed
 //
 // This handles the following potential consolidation result:
-// machines=[t3a.2xlarge, t3a.2xlarge, t3a.small] -> 1 of t3a.small, t3a.xlarge, t3a.2xlarge
+// nodes=[t3a.2xlarge, t3a.2xlarge, t3a.small] -> 1 of t3a.small, t3a.xlarge, t3a.2xlarge
 //
 // In this case, we shouldn't perform this consolidation at all.  This is equivalent to just
-// deleting the 2x t3a.xlarge machines.  This code will identify that t3a.small is in both lists and filter
+// deleting the 2x t3a.xlarge nodes.  This code will identify that t3a.small is in both lists and filter
 // out any instance type that is the same or more expensive than the t3a.small
 //
 // For another scenario:
-// machines=[t3a.2xlarge, t3a.2xlarge, t3a.small] -> 1 of t3a.nano, t3a.small, t3a.xlarge, t3a.2xlarge
+// nodes=[t3a.2xlarge, t3a.2xlarge, t3a.small] -> 1 of t3a.nano, t3a.small, t3a.xlarge, t3a.2xlarge
 //
 // This code sees that t3a.small is the cheapest type in both lists and filters it and anything more expensive out
 // leaving the valid consolidation:
-// machines=[t3a.2xlarge, t3a.2xlarge, t3a.small] -> 1 of t3a.nano
-func filterOutSameType(newMachine *scheduling.NodeClaim, consolidate []*Candidate) []*cloudprovider.InstanceType {
-	existingInstanceTypes := sets.NewString()
+// nodes=[t3a.2xlarge, t3a.2xlarge, t3a.small] -> 1 of t3a.nano
+func filterOutSameType(newNode *scheduling.NodeClaim, consolidate []*Candidate) []*cloudprovider.InstanceType {
```

Comment:
nit: match the variable name
```suggestion
func filterOutSameType(newNodeClaim *scheduling.NodeClaim, consolidate []*Candidate) []*cloudprovider.InstanceType {
```